version.go

```diff
@@ -107,20 +110,30 @@ func (v *Version) Compare(other *Version) int {
 	}
 
 	// Compare the segments
-	for i := 0; i < len(segmentsSelf); i++ {
-		lhs := segmentsSelf[i]
+	// Because a constraint could have more/less specificity than the version it's
+	// checking, we need to account for a lopsided or jagged comparison
+	var matched int
+	for i, lhs := range segmentsSelf {
+		if i > len(segmentsOther)-1 {
+			// The constraint only had X places, but the version we're checking had
+			// X+1. This means our version is greater than the constraint version we're
+			// checking , ex: 1.2.3.4 > 1.2.3
+			matched = 1
+			break
+		}
 		rhs := segmentsOther[i]
-
 		if lhs == rhs {
 			continue
 		} else if lhs < rhs {
-			return -1
+			matched = -1
+			break
 		} else {
-			return 1
+			matched = 1
+			break
 		}
 	}
 
-	panic("should not be reached")
+	return matched
```

Comment:
I got rid of this for 2 reasons:

1. I don't want to be around when this edge case managed to happen :)
2. You really should never panic in a client library